### PR TITLE
Libpeas single r1

### PR DESCRIPTION
**Pull request by SoapGentoo**
```
--- libpeas-1.20.0.ebuild
+++ libpeas-1.20.0-r1.ebuild
@@ -3,19 +3,19 @@
 
 EAPI=6
 GNOME2_LA_PUNT="yes"
-PYTHON_COMPAT=( python{2_7,3_4,3_5} )
+PYTHON_COMPAT=( python{3_4,3_5} )
 
-inherit autotools eutils gnome2 multilib python-r1 virtualx
+inherit autotools eutils gnome2 multilib python-single-r1 virtualx
 
 DESCRIPTION="A GObject plugins library"
 HOMEPAGE="https://developer.gnome.org/libpeas/stable/"
 
 LICENSE="LGPL-2+"
 SLOT="0"
-KEYWORDS="~alpha amd64 ~arm ~ia64 ~ppc ~ppc64 ~sparc x86 ~amd64-fbsd ~amd64-linux ~x86-linux"
+KEYWORDS="~alpha ~amd64 ~arm ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-fbsd ~amd64-linux ~x86-linux"
 
 IUSE="+gtk glade jit lua +python"
-REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} ?? ( $(python_gen_useflags 'python3*') ) )"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
 
 RDEPEND="
 	>=dev-libs/glib-2.38:2
@@ -40,6 +40,10 @@
 "
 # eautoreconf needs gobject-introspection-common, gnome-common
 
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
 src_prepare() {
 	# Gentoo uses unversioned lua - lua.pc instad of lua5.1.pc, /usr/bin/lua instead of /usr/bin/lua5.1
 	eapply "${FILESDIR}"/${PN}-1.14.0-lua.pc.patch
@@ -55,32 +59,15 @@
 		$(use_enable gtk)
 		--disable-static
 
-		# possibly overriden below
-		--disable-python{2,3}
-		--disable-lua5.1
-		--disable-luajit
+		# py2 not supported anymore
+		--disable-python2
+		$(use_enable python python3)
+
+		# lua
+		$(use_enable lua lua5.1)
+		$(use_enable $(usex jit lua jit) luajit)
 	)
 
-	if use lua ; then
-		if use jit ; then
-			myconf+=( --enable-lua5.1 --enable-luajit )
-		else
-			myconf+=( --enable-lua5.1 )
-		fi
-	fi
-
-	python_configure() {
-		local v
-		python_is_python3 && v=3 || v=2
-		myconf+=(
-			"--enable-python${v}"
-			# it is just 'PYTHON' for py3 in the build system
-			"PYTHON${v#3}=${PYTHON}"
-			"PYTHON${v}_CONFIG=$(python_get_PYTHON_CONFIG)"
-		)
-	}
-	use python && python_foreach_impl python_configure
-
 	gnome2_src_configure "${myconf[@]}"
 }
 
```

### CODE REVIEW

File: dev-libs/libpeas/libpeas-1.20.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+			myconf+=( --enable-luajit )
+		fi
+	fi
+
+	if use python ; then
+		myconf+=(
+			--enable-python3
+			# it is just 'PYTHON' for py3 in the build system
+			PYTHON="${PYTHON}"
+			PYTHON3_CONFIG="$(python_get_PYTHON_CONFIG)"
```
Comment (mgorny) [inline dev-libs/libpeas/libpeas-1.20.0-r1.ebuild]: Could you check whether this is still necessary? I think python wrapping should be able to deal with it, now that we have one impl only. Then you could just put `$(use_enable python python3)` above.

File: media-sound/rhythmbox/rhythmbox-3.4.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	media-plugins/gst-plugins-soup:1.0
+	media-plugins/gst-plugins-libmms:1.0
+	|| (
+		media-plugins/gst-plugins-cdparanoia:1.0
+		media-plugins/gst-plugins-cdio:1.0 )
+	media-plugins/gst-plugins-meta:1.0
+	media-plugins/gst-plugins-taglib:1.0
+	x11-themes/adwaita-icon-theme
+	python? (
+		>=dev-libs/libpeas-0.7.3[${PYTHON_USEDEP}]
```
Comment (EvaSDK) [inline media-sound/rhythmbox/rhythmbox-3.4.1-r1.ebuild]: looks like we forgot about this a couple of times but we need `libpeas[python]` for the rest to be useful.

**Comment by SoapGentoo** [comment]
@nE0sIghT are the **python-r1** -> **python-single-r1** changes ok for you? This will significantly ease the burden of the libpeas ecosystem, as the current situation forces us to use horrible `^^ ( $(python_gen_useflags '*') )` constructs that need to be solved in `package.use`. Are you okay with stabilising the latest version of liferea soon, so we can clean out the non-py3 stuff?

File: media-sound/rhythmbox/rhythmbox-3.4.1.ebuild [hunk 3]
```diff
@@ -45,7 +45,10 @@ COMMON_DEPEND="
 	gnome-keyring? ( >=app-crypt/libsecret-0.18 )
 	libnotify? ( >=x11-libs/libnotify-0.7.0 )
 	lirc? ( app-misc/lirc )
-	python? ( >=dev-python/pygobject-3.0:3[${PYTHON_USEDEP}] )
+	python? (
+		>=dev-libs/libpeas-0.7.3[python,${PYTHON_USEDEP}]
```
Comment (EvaSDK) [inline media-sound/rhythmbox/rhythmbox-3.4.1.ebuild]: Actually this one can stay in RDEPEND as libpeas does not provide a pkgconfig for python support so it is not used at build time.

**Comment by nE0sIghT** [comment]
>  are the python-r1 -> python-single-r1 changes ok for you?

Yes

> Are you okay with stabilising the latest version of liferea soon, so we can clean out the non-py3 stuff?

I do not want to stabilize RC package. However I do not experience any issues with it so we can go this way only if you **really** need it.

**Comment by SoapGentoo** [comment]
@nE0sIghT I understand, but the current libpeas issue with its awful support for one-of-py3 + py2 makes it extremely cumbersome to maintain and keep python impls in sync, hence I'd rather get rid of all py2 rdeps of libpeas, with `net-news/liferea-1.10.19-r1` being one of them. Alternatively, instead of stabilising the new one, we could mask the older one and only keep an ~arch ebuild around for the time. Your ideas?

**Comment by nE0sIghT** [comment]
@SoapGentoo 
>  I'd rather get rid of all py2 rdeps of libpeas, with net-news/liferea-1.10.19-r1 being one of them

I will fill stabilization request for RC today then

**Pull request closed without merging**